core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1079,6 +1088,22 @@ public boolean isForNewerHudson() {
             }
         }
 
+        /**
+         * Returns true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @return true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @since TODO
+         */
+        public boolean isForNewerJava() {
+            try {
+                return requiredJava != null && new VersionNumber(requiredJava).isNewerThan(
+                        new VersionNumber(System.getProperty("java.version")));
+            } catch (NumberFormatException nfe) {
+                return false; // plugin doesn't declare a minimum Java version
```

Comment:
Huh? This should be logged as a `WARNING`.

Reply:
I really don't see how this could happen, but basically copied it from the existing code, which behaves identically.

Reply:
I agree with @jglick this looks wrong. The comment is misleading: this will enter the catch block iff `minimumJavaVersion` is not null, and hence specified.

Then would enter the block if `new VersionNumber(minimumJavaVersion)` throws a `NumberFormatException`. So if a minimumJavaVersion **is** set AND cannot be parsed: this should very likely be reported or I'm missing something.